edgelet/iotedge/src/check/checks/certificates_quickstart.rs

```diff
@@ -74,40 +74,49 @@ impl CertificatesQuickstart {
         });
         self.device_ca_cert_path = check.device_ca_cert_path.clone();
 
-        if settings.certificates().device_cert().is_none() {
-            let certificate_info = CertificateValidity::parse(
-                "Device CA certificate".to_owned(),
-                check.device_ca_cert_path.clone().unwrap(),
-            )?;
-            let not_after = certificate_info.not_after;
-            self.certificate_info = Some(certificate_info);
+        let certificate_info = CertificateValidity::parse(
+            "Device CA certificate".to_owned(),
+            check.device_ca_cert_path.clone().unwrap(),
+        )?;
```

Comment:
This parse call is implemented here:
https://github.com/Azure/iotedge/blob/release/1.1/edgelet/iotedge/src/check/checks/identity_certificate_expiry.rs#L85

Are there any concerns that the parse will fail for certain production certificates? The relevant call the parse function uses is `stack_from_pem`. As long as it is a pem it should be fine I think.
https://docs.rs/openssl/0.10.23/openssl/x509/struct.X509.html